apps/projects/app/components/Content/Settings.js

```diff
@@ -470,38 +483,48 @@ const SettingsMain = styled.div`
   align-items: stretch;
 `
 const SettingsFunding = styled.div`
-  display: flex;
-  flex-direction: ${({ layoutName })=> layoutName === 'small' ? 'column' : 'row'};
-> * {
-  width: 50%;
-  padding-right: 20px;
-}
-`
-const StyledNumberFormat = styled(NumberFormat)`
-  border-radius: 3px;
-  border: 1px solid #e6e6e6;
-  box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.06);
-  font-size: 16px;
-  height: 40px;
-  line-height: 1.5;
-  margin-right: 10px;
-  padding: 0 10px;
-  text-align: right;
-  width: 110px;
-`
-// https://stackoverflow.com/questions/3790935/can-i-hide-the-html5-number-input-s-spin-box
-
-// TODO: Refactor styles
-const StyledTextInput = styled(TextInput).attrs({
-  type: 'text',
-})`
-  height: 40px;
-  width: 185px;
-  font-size: 16px;
+  display: grid;
+  grid-gap: ${3 * GU}px;
+  grid-template-columns: ${({ layoutName }) => layoutName === 'small' ? '1fr' : '1fr 1fr'};
+  margin-bottom: ${3 * GU}px;
 `
+const StyledNumberFormat = ({ className, theme, ...props }) => (
```

Comment:
> only styled components can be extended by other styled components

That's not true. The `className` convention allows any component to be wrapped with `styled`. You'll see that I wrapped `StyledNumberFormat` throughout this change, using the `css` prop.

That said, I don't really like the name of this component, but I didn't think it would strengthen this PR to change the name of it. By keeping the name and the signature the same, I was able to keep unrelated code (the places that use the component) unchanged. I value that more than I value a convention around when to use `Styled` as a prefix.

> Also why don't just use a style component for this one? I don't like to see so much inlined CSS in the components.

This is related to keeping the columns a consistent width. On fields like Base Rate, we want the "dropdown part" to be as wide as its content—"ETH" vs "Autarkian Dollars"—and we want the numeric part to take up the rest of the space. This means we want to give the numeric part `flex-grow: 1` (or similar), and give `flex-grow: 0` to the dropdown. This `flex-grow: 1` on the numeric part fights with the `padding: 0 10px`, and the whole number+dropdown ends up extending beyond the right boundary of its column by 20px.

The only fix I could come up with, after a lot of trying other things, is to put the number input inside a div. I liked that I was able to keep the function signature the same as the previous single-element component.

As far as using named components vs inline `css` props: I used to be with you, that named components are preferrable. After reviewing lots of PRs from Radek where he followed A1's lead in using more inline `css` props, I stopped caring as much. Naming each of these sub-components seems at least as confusing as having inline `css` props. What would they be named?

> Another good reason, if the other 2 are not enough, is that inlined css is not easily lintable, making it error prone, while styled-components are currently being automatically linted in our project config (with stylelint)

Ah, it looks like this is, indeed, [still an issue](https://github.com/styled-components/stylelint-processor-styled-components/issues/270). Bummer. 

I do find this compelling.

I'll spend a couple minutes trying to come up with decently explanatory names for the sub-components within this one, so we can get linting on them.

I don't think it's worth spending more than a couple minutes on, though. Eventually, stylelint will probably work around this limitation.

Reply:
Sorry I was confused about extending styles from previous styled-components versions Chad.
My review on this was a bit excessive, I found your comments fair, even if the linting reason was compelling, after revisiting this, I think it is not enough to really spent lot of time on it.

I changed my mind after reading your explanation, so now I find the current code is correct. I agree with you, we can perfectly live with this `stylelint` limitation, just being a bit extra careful when dealing with inlined styles.